File: android/app/src/main/java/jagtheora/theora/Frame.java
package jagtheora.theora;

import jagtheora.misc.SimplePeer;

public final class Frame extends SimplePeer {

    public final int b;

    public final int a;

    public final int[] pixels;

    protected void init() {}

    public Frame(int arg0, int arg1) {
        this.b = arg0;
        this.a = arg1;
        this.pixels = new int[this.a * this.b];
    }

    protected void clear() {}
}

File: src/jvmMain/kotlin/jagtheora/theora/Frame.kt
package jagtheora.theora

import jagtheora.misc.SimplePeer
import kotlin.jvm.JvmField

actual class Frame actual constructor(@JvmField actual val a: Int, @JvmField actual val b: Int) : SimplePeer() {

    actual val pixels: IntArray = IntArray(this.a * this.b)

    actual external override fun clear()

    actual companion object {
        init {
            init()
        }

        /*protected*/ actual external fun init()
    }
}
